Sources/MapboxNavigation/SpriteRepository.swift

```diff
@@ -4,79 +4,93 @@ import MapboxCoreNavigation
 import MapboxDirections
 
 class SpriteRepository {
-    // Caching the Sprite and single shield icon images.
-    let spriteCache = ImageCache()
-    // Caching the single legacy shield icon images.
-    let legacyCache = ImageCache()
-    // Caching the metadata info for Sprite.
-    let infoCache =  SpriteInfoCache()
     let baseURL: URL = URL(string: "https://api.mapbox.com/styles/v1")!
-    var styleURI: StyleURI = .navigationDay
+    let defaultStyleURI: StyleURI = .navigationDay
     fileprivate(set) var imageDownloader: ReentrantImageDownloader
     
+    let requestCache: URLCaching
+    let derivedCache: BimodalImageCache
     static let shared = SpriteRepository.init()
     
     private let requestTimeOut: TimeInterval = 10
     
-    var styleID: String? {
-        styleURI.rawValue.components(separatedBy: "styles")[safe: 1]
-    }
-    
     var sessionConfiguration: URLSessionConfiguration {
         didSet {
             imageDownloader = ImageDownloader(sessionConfiguration: sessionConfiguration)
         }
     }
     
+    var userInterfaceIdiomStyles = [UIUserInterfaceIdiom: StyleURI]()
+
     init() {
         sessionConfiguration = URLSessionConfiguration.default
         sessionConfiguration.timeoutIntervalForRequest = self.requestTimeOut
         imageDownloader = ImageDownloader(sessionConfiguration: sessionConfiguration)
+        requestCache = URLDataCache()
+        derivedCache = ImageCache()
     }
     
-    func updateStyle(styleURI: StyleURI?, completion: @escaping ImageDownloadCompletionHandler) {
+    func updateStyle(styleURI: StyleURI?,
+                     idiom: UIUserInterfaceIdiom = .phone,
+                     completion: @escaping ImageDownloadCompletionHandler) {
         // If no valid StyleURI provided, use the default Day style.
-        let newStyleURI = styleURI ?? self.styleURI
-        guard newStyleURI != self.styleURI || getSpriteImage() == nil else {
+        let newStyleURI = styleURI ?? defaultStyleURI
+        
+        if !userInterfaceIdiomStyles.values.contains(newStyleURI) {
+            updateUserInterfaceIdiomStyles(styleURI: newStyleURI, idiom: idiom)
```

Comment:
`updateUserInterfaceIdiomStyles(styleURI: newStyleURI, idiom: idiom)` is called in both cases, can it be moved outside of condition check?